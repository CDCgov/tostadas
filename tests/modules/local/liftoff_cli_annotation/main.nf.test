nextflow_process {

    name "Test Process LIFTOFF_CLI"
    script "modules/local/liftoff_cli_annotation/main.nf"
    process "LIFTOFF_CLI"

    test("Should run without failures") {

        when {
            params {
                // parameter values
                output_dir = "${outputDir}"
                overwrite_output = true
                lift_unmapped_features_file_name = "output.unmapped_features.txt"
                lift_coverage_threshold = 0.5
                lift_child_feature_align_threshold = 0.5
                lift_distance_scaling_factor = 2.0
                lift_flank = 0
                lift_parallel_processes = 8
                lift_feature_types = "${projectDir}/assets/feature_types.txt"
                lift_copy_threshold = 1.0
                lift_overlap = 0.1
                lift_mismatch = 2
                lift_gap_open = 2
                lift_gap_extend = 1
            }
            process {
                """
                input[0] = ["NY0006", file("${projectDir}/assets/sample_metadata/mpxv_test_metadata.xlsx"), 
                                         file("${projectDir}/assets/sample_fastas/mpox/NY0006.fasta"), 
                                         file("${projectDir}/assets/sample_fastqs/mpox/2022-028-7666_S3_L001_R1_001.fastq.gz"), 
                                         file("${projectDir}/assets/sample_fastqs/mpox/2022-028-7666_S3_L001_R2_001.fastq.gz")]
                input[1] = file("${projectDir}/assets/ref/ref.MPXV.NC063383.v7.fasta")
                input[2] = file("${projectDir}/assets/ref/ref.MPXV.NC063383.v7.gff")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }
    }
}

