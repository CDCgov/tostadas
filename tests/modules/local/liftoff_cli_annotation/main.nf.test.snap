{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006.fasta:md5,4a11c4fe6d828bf5e411c21d10075b7c"
                    ]
                ],
                "1": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006.liftoff-orig.gff:md5,5c311a89c5ce7e828ba6fc6477da7cf4"
                    ]
                ],
                "2": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "output.unmapped_features.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "errors": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "output.unmapped_features.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "fasta": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006.fasta:md5,4a11c4fe6d828bf5e411c21d10075b7c"
                    ]
                ],
                "gff": [
                    [
                        {
                            "batch_id": "batch_1",
                            "batch_tsv": "batch_1.tsv:md5,4ab00a8f04e89e82dd5504f5ca0c0a6c",
                            "sample_id": "NY0006"
                        },
                        "NY0006.liftoff-orig.gff:md5,5c311a89c5ce7e828ba6fc6477da7cf4"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:21:57.639675966"
    }
}